### PR TITLE
MultiSlider tests

### PR DESCRIPTION
**Pull request by giladgray**
refactor all Slider tests to be much more sensible.

- new `Handle` test suite tests onChange/onRelease and keyboard/mouse/touch events and vertical 😮 
- `Slider` and `RangeSlider` only test their added props and specific handle setup
- `MultiSlider` has several sub-suites for handles, labels, track, validation

this will unblock #2569

### CODE REVIEW

File: packages/core/src/components/slider/handle.tsx [hunk 1]
```diff
@@ -153,9 +154,9 @@ export class Handle extends AbstractPureComponent<IHandleProps, IHandleState> {
     private handleMoveEndedAt = (clientPixel: number) => {
         this.removeDocumentEventListeners();
         this.setState({ isMoving: false });
-        // not using changeValue because we want to invoke the handler regardless of current prop value
+        // always invoke onRelease; changeValue may call onChange if value is different
```
Comment (giladgray) [inline packages/core/src/components/slider/handle.tsx]: this is a small API improvement. was weird that _only_ onRelease was invoked in this case.

File: packages/core/src/components/slider/slider.tsx [hunk 2]
```diff
@@ -48,9 +48,9 @@ export class Slider extends AbstractPureComponent<ISliderProps> {
                 <SliderHandle
                     value={value}
                     intentAfter={value < initialValue ? Intent.PRIMARY : undefined}
-                    intentBefore={value > initialValue ? Intent.PRIMARY : undefined}
+                    intentBefore={value >= initialValue ? Intent.PRIMARY : undefined}
                 />
-                <SliderHandle value={0} interactionKind="none" />
+                <SliderHandle value={initialValue} interactionKind="none" />
```
Comment (giladgray) [inline packages/core/src/components/slider/slider.tsx]: my tests caught a bug!

File: packages/core/test/slider/multiSliderTests.tsx [hunk 3]
```diff
@@ -32,342 +35,185 @@ describe("<MultiSlider>", () => {
...
-        slider
-            .find(Handle)
-            .last()
-            .simulate("touchstart", { changedTouches: [{ clientX: tickSize * 10 }] });
-        Utils.touchMoveHorizontal(-tickSize, 5, tickSize * 10);
-        // called 3 times for the move to 9, 8, 7, and 6
-        assert.equal(onChange.callCount, 4);
-        assert.deepEqual(onChange.args.map(arg => arg[0]), [[0, 5, 9], [0, 5, 8], [0, 5, 7], [0, 5, 6]]);
-    });
+        it("moving mouse on the first handle updates the first value", () => {
```
Comment (giladgray) [inline packages/core/test/slider/multiSliderTests.tsx]: deleted touch tests here as they're "handled" by `Handle` tests

File: packages/core/test/slider/multiSliderTests.tsx [hunk 4]
```diff
@@ -32,342 +35,185 @@ describe("<MultiSlider>", () => {
...
-    it("disabled slider does not respond to touch movement", () => {
-        const slider = renderSlider({ disabled: true, onChange });
-        slider
-            .find(Handle)
-            .first()
-            .simulate("touchstart", { changedTouches: [{ clientX: 0 }] });
-        Utils.touchMoveHorizontal(slider.state("tickSize"), 5);
-        assert.isTrue(onChange.notCalled, "onChange was called when disabled");
-    });
+    describe("labels", () => {
```
Comment (giladgray) [inline packages/core/test/slider/multiSliderTests.tsx]: this block moved from old `sliderTests`

File: packages/core/test/slider/rangeSliderTests.tsx [hunk 5]
```diff
@@ -11,312 +11,51 @@ import * as sinon from "sinon";
...
+        assert.lengthOf(track, 1);
+        assert.equal(track.getDOMNode().getBoundingClientRect().width, STEP_SIZE * 3);
     });
 
     it("throws error if range value contains null", () => {
         expectPropValidationError(RangeSlider, { value: [null, 5] });
         expectPropValidationError(RangeSlider, { value: [100, null] });
     });
 
-    it("moving mouse on left handle updates first value in range", () => {
```
Comment (giladgray) [inline packages/core/test/slider/rangeSliderTests.tsx]: no need to re-test handle logic now

File: packages/core/test/slider/sliderTestUtils.ts [hunk 6]
```diff
@@ -6,77 +6,74 @@
...
+export const DRAG_SIZE = 20;
 
-export function getSliderBottomPixel(slider: ReactWrapper<ISliderBaseProps, any>) {
-    const { height, top } = slider.getDOMNode().getBoundingClientRect();
-    return height + top;
+/**
+ * Simulates a full move of a slider handle: engage, move, release.
+ * Supports touch and vertical events. Use options to configure exact movement.
+ */
+export function simulateMovement(wrapper: ReactWrapper, options: IMoveOptions) {
```
Comment (giladgray) [inline packages/core/test/slider/sliderTestUtils.ts]: one util to rule them all! 💍

File: packages/core/src/components/slider/slider.tsx (hunk 2 above)
Comment (jscheiny) [reply packages/core/src/components/slider/slider.tsx]: Good catch!

**Pull request merged**